gradle-consistent-versions-idea-plugin/src/test/java/com/palantir/gradle/versions/intellij/VersionPropsLexerTests.java

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.hash.Hashing;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiRecursiveElementVisitor;
+import com.intellij.testFramework.fixtures.JavaCodeInsightTestFixture;
+import com.intellij.testFramework.fixtures.LightJavaCodeInsightFixtureTestCase5;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class VersionPropsLexerTests extends LightJavaCodeInsightFixtureTestCase5 {
+
+    @Override
+    protected final String getRelativePath() {
+        return "";
+    }
+
+    @Override
+    protected final String getTestDataPath() {
+        return "";
+    }
+
+    @BeforeAll
+    public static void before() throws IOException {
+        Optional<String> inCi = Optional.ofNullable(System.getenv("CI"));
+
+        if (!inCi.equals(Optional.of("true"))) {
+            Path directoryToDelete = Paths.get("src/test/resources/lexerTests");
+            if (Files.exists(directoryToDelete)) {
+                deleteDirectoryRecursively(directoryToDelete);
+            }
+        }
+    }
+
+    private static void deleteDirectoryRecursively(Path path) throws IOException {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(path)) {
+            for (Path entry : stream) {
+                if (Files.isDirectory(entry)) {
+                    deleteDirectoryRecursively(entry);
+                } else {
+                    Files.delete(entry);
+                }
+            }
+        }
+        Files.delete(path);
+    }
+
+    @ValueSource(
+            strings = {
+                "a.normal:example = 1",
+                "random.characters:after = version blah blah",
+                "a.normal:example = 1",
+                "has.end.of:line=comments#here is the comment",
+                "odd/.characters@?'.in!:all\\\\/ = the_pl4ces.,?",
+                "has.end.of:line=comments # here is the comment",
+                " has.extra.space:bad_token = 1",
+                "has . space . around:dots = 1",
+                "# A comment",
+                "has.spaces.around : colon = 1",
+                "has.no.spaces.around:equals=1",
+                "has.loads.of.spaces:around   =   equals",
+                "    # comment after spaces"
+            })
```

Comment:
There's one part which is missing: if you were to change one of these tests, you would never automatically delete the existing test. I think we can do this though: change this `@ValueSource` into a `@MethodSource`, have a method that returns all these test cases. Then make one `@Test` in this file that:

1. Reads all the test cases from the method
1. If on CI: errors if there are any extra tests in the `lexerTests` directory, telling people to run the test locally to fix (can use `describedAs` in assertj).
2. If not on CI: just deletes the extra tests in the `lexerTests` directory

Reply:
so I think I have done this, now a set of tests is passed in and we check if the directory is in the expected form